sds/src/match_validation/validator_utils.rs

```diff
@@ -14,7 +14,7 @@ pub fn new_match_validator_from_type(config: &MatchValidationType) -> Box<dyn Ma
             _ => panic!("This aws type shall not be used to create a validator"),
         },
         MatchValidationType::CustomHttp(http_config) => {
-            Box::new(HttpValidator::new(http_config.clone()))
+            Box::new(HttpValidator::new_clone_config(http_config))
```

Comment:
I also think that this method should returned a `Result<Box<dyn MatchValidator>, MatchValidationError>` similar to the [`RegexValidationError`](https://github.com/DataDog/dd-sensitive-data-scanner/blob/f0325a5e0eaf2ecdd9b899bf9e7e69a9e7d43895/sds/src/validation.rs#L7) in this [`CreateScannerError`](https://github.com/DataDog/dd-sensitive-data-scanner/blob/bbe8040fae6452d66efe4dec20c957d6f3e794a4/sds/src/scanner/error.rs#L20). In fact, we don't want the scanner code to panic but rather to return errors to the user.

Same as for the previous comment, it can be tackled in a follow-up PR.

Reply:
I disagree here, the new_clone_config will never panic it just clones the configuration data

Reply:
I mean, I think the method `new_match_validator_from_type` should return a result instead of panic

Reply:
Oh my bad I misunderstood your point. I agree will add a Result

Reply:
I did it let me know what you think